### PR TITLE
MODINVSTOR-626 Update bulk download endpoint to save holdings UUIDs

### PR DESCRIPTION
**Pull request by andrii-novytskyi**
**PURPOSE:**

https://issues.folio.org/browse/MODINVSTOR-626
Expand the bulk/ids endpoint to be able to get holdings uuids.

**APPROACH:**
1. Added query field recordType. If no record type, then INSTANCE type will be used to search UUIDs.
2. Change url from instance-bulk/ids to record-bulk/ids, permission name, and so on.
3. Added test case for HOLDINGS record type.

**TODO:**
Should be merged together with PRs for below modules:
- [x]  UI-inventory - {https://github.com/folio-org/ui-inventory/pull/1242 }
- [x] data-export - {https://github.com/folio-org/mod-data-export/pull/220 }   
- [x] perf-testing - {https://github.com/folio-org/perf-testing/pull/34 }
- [x] ui-agreeements - {https://github.com/folio-org/ui-agreements/pull/687 }
- [x] ui-plugin-create-inventory-record - {https://github.com/folio-org/ui-plugin-create-inventory-records/pull/43 }

### CODE REVIEW

**Review by illia-borysenko — APPROVED** [review]

**Review by Igor-Gorchakov — APPROVED** [review]

File: descriptors/ModuleDescriptor-template.json
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json:496]: This is a breaking change, once you merge this UI will not be able to download any UUIDs, block this PR and let UI devs make necessary changes and only then merge this PR

File: descriptors/ModuleDescriptor-template.json
Comment (KVupp) [inline descriptors/ModuleDescriptor-template.json:491]: I am not sure if renaming an interface was done previously in FOLIO, but we need to increment the version as well

Comment (KVupp) [reply descriptors/ModuleDescriptor-template.json:491]: Also verify if this interface or API is used anywhere else in FOLIO other than the ui-inventory, we need to make those changes in that modules as well

File: ramls/record-bulk.raml
Comment (KVupp) [inline ramls/record-bulk.raml:9]: just for readability sake, please add what records can be downloaded and that ID is infact the UUID of the records

File: ramls/record-bulk.raml
Comment (KVupp) [inline ramls/record-bulk.raml:45]:
```suggestion
          description: Determines which record type UUIDs should be fetched
```

File: ramls/record-bulk.raml
Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:34]: yes, if you will take a look at the test, you will see it

File: descriptors/ModuleDescriptor-template.json
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json:496]: @KVupp yes, I described it in notes

File: descriptors/ModuleDescriptor-template.json
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json:491]: UI-inventory, data-export and perf-testing are using this interface

Comment (KVupp) [reply descriptors/ModuleDescriptor-template.json:491]: Thanks! Let's add this list as TODO and when PRs are ready tag them here

File: ramls/record-bulk.raml
Comment (KVupp) [inline ramls/record-bulk.raml:2]: change version here as well

Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:2]: @KVupp updated

File: descriptors/ModuleDescriptor-template.json
Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json:491]: updated

File: ramls/record-bulk.raml
Comment (KVupp) [reply ramls/record-bulk.raml:2]: and correct the base URI as well please

File: ramls/record-bulk.raml
Comment (KVupp) [inline ramls/record-bulk.raml:47]: A minor one but we have singular for instance but plural for holdings, would be easier to make it consistent

File: ramls/record-bulk.raml
Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:2]: done

**Review by KVupp — APPROVED** [review]

File: ramls/record-bulk.raml
Comment (andrii-novytskyi) [reply ramls/record-bulk.raml:47]: for some reason everywhere in folio holdings are used instead of holding, but ok, I will change to holding

Comment (andrii-novytskyi) [reply ramls/record-bulk.raml]: @KVupp  done

File: descriptors/ModuleDescriptor-template.json
Comment (marcjohnson-kint) [inline descriptors/ModuleDescriptor-template.json:491]: Given that this interface is being renamed, please could it be given a name that provides more context than only the term `record`.

What records are involved? If it is any inventory record, could `inventory-record-bulk` be an acceptable name?

This name takes up a very general space and could easily conflict with other interface names that are not related to inventory at all.

Comment (andrii-novytskyi) [reply descriptors/ModuleDescriptor-template.json]: make sense, updated, thanks!

**Pull request merged**
